Branch-SDK/src/io/branch/referral/RemoteInterface.java

```diff
@@ -156,6 +158,16 @@ private ServerResponse make_restful_get(String baseUrl, String tag, int timeout,
             timeout = DEFAULT_TIMEOUT;
         }
         if (addCommonParams(getParameters, retryNumber)) {
+            if (params != null) {
+                Iterator keys = params.keys();
+                while (keys.hasNext()) {
```

Comment:
Is there a risk that the request will be modified while iterating? we had a nasty bug for a while because we did iteration here and still allowed the server requests to be modified.